src/doc/reference.md

```diff
@@ -1237,6 +1237,9 @@ the guarantee that these issues are never caused by safe code.
 * Breaking the [pointer aliasing
   rules](http://llvm.org/docs/LangRef.html#pointer-aliasing-rules)
   with raw pointers (a subset of the rules used by C)
+* `&mut` and `&` follow LLVM’s scoped [noalias] model, except if the `&T`
```

Comment:
There's a point just above talking about `UnsafeCell`, maybe these could be merged/moved closer together?